File: scala/src/algorithms/warmup/PlusMinus.scala
package algorithms.warmup

object PlusMinus {
  def main(args: Array[String]): Unit = {
    val n = scala.io.StdIn.readLine().toInt
    val a = scala.io.StdIn.readLine().split(" ").map(_.toInt)

    var pos, neg, zero = 0
    for (x <- a) {
      if (x > 0) pos += 1
      else if (x < 0) neg += 1
      else zero += 1
    }

    println(pos.toFloat / n)
    println(neg.toFloat / n)
    println(zero.toFloat / n)
  }
}

File: go/algorithms/warmup/plus_minus.go
package main

import (
	"fmt"
	"os"
	"bufio"
	"strings"
	"strconv"
)

func ConvertToInt(sa []string) ([]int) {
	si := make([]int, 0, len(sa))
	for _, a := range sa {
		i, err := strconv.Atoi(a)
		if err != nil {
			panic(err)
		}
		si = append(si, i)
	}

	return si
}

func main() {
	var n int
	fmt.Scanf("%v\n", &n)
	var s = bufio.NewScanner(os.Stdin)
	s.Scan()
	var a = ConvertToInt(strings.Split(s.Text(), " "))

	var pos, neg, zero = 0, 0, 0
	for x := 0; x < n; x++ {
		if a[x] > 0 {
			pos++
		} else if a[x] < 0 {
			neg++
		} else {
			zero++
		}
	}

	fmt.Println(float32(pos) / float32(n))
	fmt.Println(float32(neg) / float32(n))
	fmt.Println(float32(zero) / float32(n))
}
